api/src/main/java/io/minio/messages/ResponseDate.java

```diff
@@ -35,6 +35,9 @@ public class ResponseDate {
   public static final DateTimeFormatter MINIO_RESPONSE_DATE_FORMAT =
       DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH':'mm':'ss'Z'", Locale.US).withZone(Time.UTC);
 
+  public static final DateTimeFormatter MINIO_RESPONSE_DATE_FORMAT_SHORT =
```

Comment:
Why don't you comply with AWS date format like "yyyy-MM-dd'T'HH':'mm':'ss'.'SSS'Z'" ?

Reply:
SSS'Z allows .S'Z , .SS'Z, .SSS'Z - this is allowed in Go time.Parse() 

I am actually fixing something else here we need to support two formats for retainUntilDate 

Reading aws-sdk-java will update the PR.